doc/devel/release_guide.rst

```diff
@@ -48,7 +48,10 @@ Review and commit changes.  Some issue/PR titles may not be valid rst (the most
 Check Docs
 ----------
 
-Before tagging, make sure that the docs build cleanly ::
+Before tagging, update the what's new listing in :file:`doc/users/whats_new.rst`
```

Comment:
I don't think it should be done in this PR, but I still think it'd be good to have some kind of "credit" system for various features. E.g. just making sure that every new feature or bugfix has a `by XXX` at the end, optionally with a link to a user webpage or something.